packages/schema/parse/src/comparison/compapers/ArrayType.ts

```diff
@@ -0,0 +1,89 @@
+import { ArrayDefinition } from "../../typeInfo";
```

Comment:
The dir should be named comparers not compapers :D
Also, in TS, it's better to either name the file by the component in it (if the components is a func: getTest then getTest.ts, if the component is a class SomeClass then SomeClass.ts (keeping the casing)

Alternatively if there are more components in a single file, which IMO should be avoided if possible. Then use dash casing: some-name.ts

For the current file that means either: array-type.ts or compareArrayType.ts
The former (array-type.ts) might be best since other files in this dir have multiple components, and there's also "compare" in the name of the dir "compare-rs" so it's kind of redundant.